code/lib/cli/src/templates.ts

```diff
@@ -0,0 +1,179 @@
+import { minVersion, validRange } from 'semver';
+import type { TemplateConfiguration } from './project_types';
+import { ProjectType } from './project_types';
+
+function ltMajor(versionRange: string, major: number) {
+  // Uses validRange to avoid a throw from minVersion if an invalid range gets passed
+  return validRange(versionRange) && minVersion(versionRange).major < major;
+}
+
+function gtMajor(versionRange: string, major: number) {
+  // Uses validRange to avoid a throw from minVersion if an invalid range gets passed
+  return validRange(versionRange) && minVersion(versionRange).major > major;
+}
+
+function eqMajor(versionRange: string, major: number) {
+  // Uses validRange to avoid a throw from minVersion if an invalid range gets passed
+  return validRange(versionRange) && minVersion(versionRange).major === major;
+}
+
+/**
+ * Configuration to match a storybook preset template.
+ *
+ * This has to be an array sorted in order of specificity/priority.
+ * Reason: both REACT and WEBPACK_REACT have react as dependency,
+ * therefore WEBPACK_REACT has to come first, as it's more specific.
+ */
+export const supportedTemplates: TemplateConfiguration[] = [
```

Comment:
I wonder if this could be combined into `code/lib/cli/src/sandbox-templates.ts` somehow? Not in this PR, but maybe in the future?